src/fm.c

```diff
@@ -8,12 +12,39 @@ int main(int argc, char const *argv[])
         fprintf(stderr,
                 "Usage: %s -p genome\n       %s genome reads\n",
                 argv[0], argv[0]);
+        fprintf(stderr,
+                "argc=%d\n",
+                argc);
         return 1;
     }
+    char* processFileName = get_file_name_by_fa(argv[2]);
+
     if (strcmp("-p", argv[1]) == 0)
     {
         // preprocessing
-        printf("Preprocessing genome %s\n", argv[2]);
+        //printf("Preprocessing genome %s\n", argv[2]);
+        FILE* processFile = get_file(processFileName);
```

Comment:
If `get_file()` returns an opened file (which I assume given the type `FILE *`), then you have to close it at some point. I don't see a `fclose(processFile)`  here.